src/context_managers.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+import asyncpg
+
+
+log = logging.getLogger(__name__)
+
+
+class Connection:
+    def __init__(self, db_uri):
+        self._db_uri = db_uri
+
+    async def __aenter__(self):
+        self._conn = await asyncpg.connect(self._db_uri)
+        return self._conn
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        if exc_type:
+            logging.info(f'Exception {exc_val} has been handled')
+        await self._conn.close()
+
+
+class Query:
+    def __init__(self, path: str, method: chr, encoding: str = 'utf-8'):
+        self._path = path
+        self._method = method
+        self._encoding = encoding
+
+    def __enter__(self):
+        return open(self._path, self._method, encoding=self._encoding)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if exc_type:
+            log.error(exc_val)
+        self._file.close()
+        return True
```

Comment:
For what reason you have implemented that context managers?
To log exceptions?

Just make a top-level try/except and make logging there, and do not redevelop the wheel